CMakeLists.txt

```diff
@@ -27,18 +27,39 @@ project(Mir)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
-set(MIR_VERSION_MAJOR 0)
-set(MIR_VERSION_MINOR 29)
-set(MIR_VERSION_PATCH 0)
+execute_process(
+  COMMAND git describe --abbrev=0
+  OUTPUT_VARIABLE MIR_VERSION_GIT
+  RESULT_VARIABLE MIR_VERSION_GIT_RESULT)
+
+if (${MIR_VERSION_GIT_RESULT} EQUAL 0)
+  string(REGEX MATCH   "([0-9]+)\\.([0-9]+)\\.([0-9]+)"       MIR_VERSION       "${MIR_VERSION_GIT}")
```

Comment:
Maybe make this more strict? `^v([0-9]+)\\.([0-9]+)\\.([0-9]+)$`?

What if it doesn't match? Maybe the `if()` should check the match as well?

Reply:
It isn't quite as simple as that: we don't want to extract the "v".

If it doesn't match expectations we'll get an obscure compilation error where the defines are used in code. I guess a format check could give a clearer error message.

Reply:
Even better, `git describe` has a `--match` option that you can pass a glob to - we only want to consider `v[0-9]*` tags, so I suppose that would be enough. It would even mean that even if we do have a non-version tag, only that one will be used in here.